### PR TITLE
Feat/loguru

### PR DESCRIPTION
**Pull request by pgierz**
Replaces all print statements with loguru statements.

### CODE REVIEW

**Comment by mandresm** [comment]
Hey @pgierz! thanks for this development! Do you have any idea why are the tests failing?

File: src/esm_runscripts/compute.py [hunk 1]
```diff
@@ -362,8 +356,7 @@ class EsmConfigDumper(yaml.dumper.Dumper):
     expid = config["general"]["expid"]
     it_coupled_model_name = config["general"]["iterative_coupled_model"]
     config_file_path = (
-        f"{thisrun_config_dir}/"
-        f"{expid}_{it_coupled_model_name}finished_config.yaml"
+        f"{thisrun_config_dir}/" f"{expid}_{it_coupled_model_name}finished_config.yaml"
```
Comment (mandresm) [inline src/esm_runscripts/compute.py]:
```suggestion
        f"{thisrun_config_dir}/{expid}_{it_coupled_model_name}finished_config.yaml"
```

**Comment by pgierz** [comment]
Looks like a "general" bug in the CI's ubuntu platform. HDF5 is not able to install...? I can have a more detailed looked after the data science seminar.

File: src/esm_runscripts/compute.py [hunk 2]
```diff
@@ -45,9 +45,9 @@ def compile_model(config):
     if not version:
         return config
     if config.get("general", {}).get("run_number") == 1:
-        print("First year, checking if we need to compile...")
+        logger.info("First year, checking if we need to compile...")
         if not config.get("general", {}).get("use_compiled_model", True):
-            print(f"Huhu --> compiling {model}-{version}")
+            logger.debug(f"Huhu --> compiling {model}-{version}")
```
Comment (nwieters) [inline src/esm_runscripts/compute.py]:
```suggestion
            logger.info(f"Huhu --> compiling {model}-{version}")
```
I think when you inform about the checking whether to compile or not you can then also inform about when it is compiling.

Comment (nwieters) [reply src/esm_runscripts/compute.py]: Is it correct, that `logger.info` will always be written to log file and `logger.debug` only in debug mode?

File: src/esm_runscripts/compute.py (hunk 1 above)
Comment (nwieters) [inline src/esm_runscripts/compute.py]:
```suggestion
        f"{thisrun_config_dir}/{expid}_{it_coupled_model_name}finished_config.yaml"
```

File: src/esm_runscripts/compute.py [hunk 3]
```diff
@@ -523,10 +516,10 @@ def copy_tools_to_thisrun(config):
     # In case there is no esm_tools or namelists in the experiment folder,
     # copy from the default esm_tools path
     if not os.path.isdir(tools_dir):
-        print("Copying standard yamls from: ", esm_tools.get_config_filepath())
+        logger.info("Copying standard yamls from: ", esm_tools.get_config_filepath())
```
Comment (nwieters) [inline src/esm_runscripts/compute.py]:
```suggestion
        logger.info(f"Copying standard yamls from: {esm_tools.get_config_filepath()}")
```

File: src/esm_runscripts/compute.py [hunk 4]
```diff
@@ -523,10 +516,10 @@ def copy_tools_to_thisrun(config):
...
         esm_tools.copy_config_folder(tools_dir)
     if not os.path.isdir(namelists_dir):
-        print(
+        logger.info(
             "Copying standard namelists from: ",
             esm_tools.get_namelist_filepath(),
         )
```
Comment (nwieters) [inline src/esm_runscripts/compute.py]:
```suggestion
        logger.info(
            f"Copying standard namelists from: {esm_tools.get_namelist_filepath()}"
        )
```

File: src/esm_runscripts/coupler.py [hunk 5]
```diff
@@ -51,7 +52,7 @@ def __init__(self, full_config, name):
             )
 
         else:
-            print("Unknown coupler :", name)
+            logger.error("Unknown coupler :", name)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
            logger.error(f"Unknown coupler : {name}")
```

File: src/esm_runscripts/coupler.py [hunk 6]
```diff
@@ -94,7 +95,9 @@ def add_files(self, full_config):
                         rights = [rightside]
 
                     if not len(lefts) == len(rights):
-                        print("Left and right side of coupling don't match: ", coupling)
+                        logger.error(
+                            "Left and right side of coupling don't match: ", coupling
+                        )
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                        logger.error(
                            f"Left and right side of coupling don't match: {coupling}"
                        )
```

File: src/esm_runscripts/coupler.py [hunk 7]
```diff
@@ -142,16 +145,16 @@ def add_files(self, full_config):
...
-                                            print(
+                                            logger.error(
                                                 "All fields coupled together need to exist on same grid"
                                             )
                                             sys.exit(0)
                                 if found_right and found_left:
                                     break
                         if not found_left:
-                            print("Coupling var not found: ", left)
+                            logger.error("Coupling var not found: ", left)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                            logger.error(f"Coupling var not found: {left}")
```

File: src/esm_runscripts/coupler.py [hunk 8]
```diff
@@ -142,16 +145,16 @@ def add_files(self, full_config):
...
                         if not found_right:
-                            print("Coupling var not found: ", right)
+                            logger.error("Coupling var not found: ", right)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                            logger.error(f"Coupling var not found: {right}")
```

File: src/esm_runscripts/coupler.py [hunk 9]
```diff
@@ -236,20 +239,20 @@ def prepare_restarts(self, full_config):
...
                     dym_issue = False
                     if len(all_lefts) != len(all_leftmodels):
-                        print(
+                        logger.error(
                             "Coupling fields and their corresponding models do not"
                             + "have the same dimensions:"
                         )
-                        print("all_lefts =", all_lefts)
-                        print("all_leftmodels =", all_leftmodels)
+                        logger.error("all_lefts =", all_lefts)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                        logger.error(f"all_lefts = {all_lefts}")
```

File: src/esm_runscripts/coupler.py [hunk 10]
```diff
@@ -236,20 +239,20 @@ def prepare_restarts(self, full_config):
...
+                        logger.error("all_leftmodels =", all_leftmodels)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                        logger.error(f"all_leftmodels = {all_leftmodels}")
```

File: src/esm_runscripts/coupler.py [hunk 11]
```diff
@@ -236,20 +239,20 @@ def prepare_restarts(self, full_config):
...
                         dym_issue = True
                     if len(all_rights) != len(all_rightmodels):
-                        print(
+                        logger.error(
                             "Coupling fields and their corresponding models do not"
                             + "have the same dimensions:"
                         )
-                        print("all_rights =", all_rights)
-                        print("all_rightmodels =", all_rightmodels)
+                        logger.error("all_rights =", all_rights)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                        logger.error(f"all_rights = {all_rights}")
```

File: src/esm_runscripts/coupler.py [hunk 12]
```diff
@@ -236,20 +239,20 @@ def prepare_restarts(self, full_config):
...
+                        logger.error("all_rightmodels =", all_rightmodels)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                        logger.error(f"all_rightmodels = {all_rightmodels}")
```

File: src/esm_runscripts/coupler.py [hunk 13]
```diff
@@ -288,7 +291,7 @@ def add_couplings(self, full_config):
                             rights = [rightside]
 
                         if not len(lefts) == len(rights):
-                            print(
+                            logger.error(
                                 "Left and right side of coupling don't match: ",
                                 coupling,
                             )
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                            logger.error(f"Left and right side of coupling don't match: {coupling}")
```

File: src/esm_runscripts/coupler.py [hunk 14]
```diff
@@ -339,20 +342,22 @@ def add_couplings(self, full_config):
...
-                                                print(
+                                                logger.error(
                                                     "All fields coupled together need to exist on same grid"
                                                 )
                                                 sys.exit(0)
                                     if found_right and found_left:
                                         break
                             if not found_left:
-                                print("Coupling var not found: ", left)
+                                logger.error("Coupling var not found: ", left)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                                logger.error(f"Coupling var not found: {left}")
```

File: src/esm_runscripts/coupler.py [hunk 15]
```diff
@@ -339,20 +342,22 @@ def add_couplings(self, full_config):
...
                             if not found_right:
-                                print("Coupling var not found: ", right)
+                                logger.error("Coupling var not found: ", right)
```
Comment (nwieters) [inline src/esm_runscripts/coupler.py]:
```suggestion
                                logger.error(f"Coupling var not found: {right}")
```

File: src/esm_runscripts/filelists.py [hunk 16]
```diff
@@ -54,15 +52,14 @@ def rename_sources_to_targets(config):
...
-                            + model,
-                            flush=True,
-                        )
-                        helpers.print_datetime(config)
+                    logger.debug(
+                        "Renaming sources to targets for filetype "
+                        + filetype
+                        + " in model "
+                        + model,
+                        flush=True,
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: Is `flush=True` an argument to `logger.debug`?

File: src/esm_runscripts/filelists.py [hunk 17]
```diff
@@ -54,15 +52,14 @@ def rename_sources_to_targets(config):
...
+                    )
+                    helpers.print_datetime(config)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py:60]: Why have you taken this also out of the verbose if statement?

File: src/esm_runscripts/filelists.py [hunk 18]
```diff
@@ -105,7 +102,9 @@ def rename_sources_to_targets(config):
                     pass
 
                 elif (not sources and in_work) or (not sources and targets):
-                    print(filetype + "_sources missing in model " + model, flush=True)
+                    logger.error(
+                        filetype + "_sources missing in model " + model, flush=True
+                    )
                     helpers.print_datetime(config)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: Is `flush=True` an argument to `logger.error`?

File: src/esm_runscripts/filelists.py [hunk 19]
```diff
@@ -152,22 +153,23 @@ def complete_targets(config):
...
 
     return config
 
 
 def complete_sources(config):
-    if config["general"]["verbose"]:
-        print("::: Complete sources", flush=True)
-        helpers.print_datetime(config)
+    logger.debug("::: Complete sources", flush=True)
+    helpers.print_datetime(config)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: Same as above?

File: src/esm_runscripts/filelists.py [hunk 20]
```diff
@@ -239,11 +241,13 @@ def choose_needed_files(config):
                         + model,
                         flush=True,
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: Is `flush=True` an argument to `logger.error`?

File: src/esm_runscripts/filelists.py [hunk 21]
```diff
@@ -239,11 +241,13 @@ def choose_needed_files(config):
...
                     )
-                    print(config[model][filetype + "_files"], flush=True)
-                    print(config[model][filetype + "_sources"], flush=True)
+                    logger.error(config[model][filetype + "_files"], flush=True)
+                    logger.error(config[model][filetype + "_sources"], flush=True)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: flush=True?

File: src/esm_runscripts/filelists.py [hunk 22]
```diff
@@ -772,8 +773,7 @@ def replace_year_placeholder(config):
 
 
 def log_used_files(config):
-    if config["general"]["verbose"]:
-        print("\n::: Logging used files", flush=True)
+    logger.debug("\n::: Logging used files", flush=True)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: flush=True?

File: src/esm_runscripts/filelists.py [hunk 23]
```diff
@@ -812,24 +811,20 @@ def log_used_files(config):
...
+                                f'{config[model][filetype + "_sources"][category]}'
+                            ),
+                        )
+                        logger.debug(
+                            (
+                                f"- target: "
+                                f'{config[model][filetype + "_targets"][category]}'
+                            ),
+                        )
+                        helpers.print_datetime(config)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py:823]: `helpers.print_datetime(config)` outside of verbose if statement?

File: src/esm_runscripts/filelists.py [hunk 24]
```diff
@@ -899,9 +894,8 @@ def resolve_symlinks(config, file_source):
         # deniz: check if file links to itself. In UNIX
         # ln -s endless_link endless_link is a valid command
         if os.path.abspath(file_source) == points_to:
-            if config["general"]["verbose"]:
-                print(f"file {file_source} links to itself", flush=True)
-                helpers.print_datetime(config)
+            logger.debug(f"file {file_source} links to itself")
+            helpers.print_datetime(config)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py:894]: outside verbose if statement?

File: src/esm_runscripts/filelists.py [hunk 25]
```diff
@@ -997,10 +981,12 @@ def copy_files(config, filetypes, source, target):
...
         if config["general"]["verbose"]:
-            print("\n\nWARNING: These files were missing:")
+            logger.warning("\n\nWARNING: These files were missing:")
             for missing_file in missing_files:
-                print(f"- missing source: {missing_files[missing_file]}", flush=True)
-                print(f"- missing target: {missing_file}", flush=True)
+                logger.warning(
+                    f"- missing source: {missing_files[missing_file]}", flush=True
+                )
+                logger.warning(f"- missing target: {missing_file}", flush=True)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: flush=True?

File: src/esm_runscripts/filelists.py [hunk 26]
```diff
@@ -1079,18 +1063,15 @@ def report_missing_files(config):
...
+            logger.warning("MISSING FILES:")
         for missing_file in config["general"]["files_missing_when_preparing_run"]:
-            print(flush=True)
-            print(
+            logger.warning(
                 f'- missing source: {config["general"]["files_missing_when_preparing_run"][missing_file]}',
-                flush=True,
             )
-            print(f"- missing target: {missing_file}", flush=True)
+            logger.warning(f"- missing target: {missing_file}", flush=True)
```
Comment (nwieters) [inline src/esm_runscripts/filelists.py]: flush=True?

File: src/esm_runscripts/namelists.py [hunk 27]
```diff
@@ -435,38 +443,35 @@ def apply_echam_disturbance(config):
...
-                        + config["general"]["experiment_scripts_dir"]
-                        + "/disturb_years.dat",
-                        "was not found",
-                    )
+                logger.warning(
+                    "WARNING: "
+                    + config["general"]["experiment_scripts_dir"]
+                    + "/disturb_years.dat",
+                    "was not found",
+                )
```
Comment (nwieters) [inline src/esm_runscripts/namelists.py]:
```suggestion
                logger.warning(
                    "WARNING: "
                    + config["general"]["experiment_scripts_dir"]
                    + "/disturb_years.dat"
                    + " was not found.",
                )
```

File: src/esm_runscripts/namelists.py [hunk 28]
```diff
@@ -435,38 +443,35 @@ def apply_echam_disturbance(config):
...
-                        "disturbance_years",
-                        disturbance_years,
-                    )
+                logger.debug("Not applying disturbance in echam namelist.")
+                logger.debug(
+                    "Current year",
+                    current_year,
+                    "disturbance_years",
+                    disturbance_years,
+                )
```
Comment (nwieters) [inline src/esm_runscripts/namelists.py]:
```suggestion
                logger.debug(f"Current year: {current_year}, disturbance_years: {disturbance_years}")
```

File: src/esm_runscripts/oasis.py [hunk 29]
```diff
@@ -154,30 +156,27 @@ def add_coupling(
...
                     )
                     sys.exit(2)
                 add_scalar = transformation["preprocessing"][pre].get(
                     "add_scalar", None
                 )
                 if not add_scalar:
-                    print(
+                    logger.error(
                         "add_scalar needs to be defined (0 or 1) for preprocessing BLASOLD",
                         flush=True,
```
Comment (nwieters) [inline src/esm_runscripts/oasis.py]: flush=True?

File: src/esm_runscripts/oasis.py [hunk 30]
```diff
@@ -154,30 +156,27 @@ def add_coupling(
...
                     )
-                    sys.exit(2)
```
Comment (nwieters) [inline src/esm_runscripts/oasis.py:170]: Why is this removed?

File: src/esm_runscripts/prepcompute.py [hunk 31]
```diff
@@ -198,32 +193,32 @@ def wait_for_iterative_coupling(config):
...
                 counter = 0
-                file = os.path.join(config['general']['experiment_couple_dir'], file_base)
+                file = os.path.join(
+                    config["general"]["experiment_couple_dir"], file_base
+                )
                 while counter < count_max:
                     counter = counter + 1
                     if os.path.isfile(file):
-                        print("File found: ", file)
+                        logger.info("File found: ", file)
```
Comment (nwieters) [inline src/esm_runscripts/prepcompute.py]:
```suggestion
                        logger.info(f"File found: {file}")
```

File: src/esm_runscripts/prepcompute.py [hunk 32]
```diff
@@ -198,32 +193,32 @@ def wait_for_iterative_coupling(config):
...
                         break
                     else:
-                        print("Waiting for file: ", file)
-                        print("Sleep for 10 seconds...")
+                        logger.info("Waiting for file: ", file)
```
Comment (nwieters) [inline src/esm_runscripts/prepcompute.py]:
```suggestion
                        logger.info(f"Waiting for file: {file}")
```

File: src/esm_runscripts/prepexp.py [hunk 33]
```diff
@@ -77,8 +77,8 @@ def copy_tools_to_thisrun(config):
     namelists_dir = f"{scriptsdir}/esm_tools/namelists"
 
     if config["general"]["verbose"]:
-        print("Started from :", fromdir)
-        print("Scripts Dir : ", scriptsdir)
+        logger.info("Started from :", fromdir)
```
Comment (nwieters) [inline src/esm_runscripts/prepexp.py]:
```suggestion
        logger.info(f"Started from : {fromdir}")
```

File: src/esm_runscripts/prepexp.py [hunk 34]
```diff
@@ -77,8 +77,8 @@ def copy_tools_to_thisrun(config):
...
+        logger.info("Scripts Dir : ", scriptsdir)
```
Comment (nwieters) [inline src/esm_runscripts/prepexp.py]:
```suggestion
        logger.info(f"Scripts Dir : {scriptsdir}")
```

File: src/esm_runscripts/prepexp.py [hunk 35]
```diff
@@ -92,10 +92,12 @@ def copy_tools_to_thisrun(config):
     # In case there is no esm_tools or namelists in the experiment folder,
     # copy from the default esm_tools path
     if not os.path.isdir(tools_dir):
-        print("Copying standard yamls from: ", esm_tools.get_config_filepath())
+        logger.info("Copying standard yamls from: ", esm_tools.get_config_filepath())
```
Comment (nwieters) [inline src/esm_runscripts/prepexp.py]:
```suggestion
        logger.info(f"Copying standard yamls from: {esm_tools.get_config_filepath()}")
```

File: src/esm_runscripts/prepexp.py [hunk 36]
```diff
@@ -92,10 +92,12 @@ def copy_tools_to_thisrun(config):
...
         esm_tools.copy_config_folder(tools_dir)
     if not os.path.isdir(namelists_dir):
-        print("Copying standard namelists from: ", esm_tools.get_namelist_filepath())
+        logger.info(
+            "Copying standard namelists from: ", esm_tools.get_namelist_filepath()
```
Comment (nwieters) [inline src/esm_runscripts/prepexp.py]:
```suggestion
            f"Copying standard namelists from: {esm_tools.get_namelist_filepath()}"
```

File: src/esm_runscripts/tidy.py [hunk 37]
```diff
@@ -147,46 +147,48 @@ def _clean_run_determine_user_choice(config):
...
-            print("    clean_runs: ", user_clean)
-            print()
-            print("Please only use one of these!")
-            print("------------------------------------------")
+            logger.error("------------------------------------------")
+            logger.error("You have set both in your config:")
+            logger.error()
+            logger.error("general:")
+            logger.error(
+                "    clean_this_rundir: ", config["general"]["clean_this_rundir"]
```
Comment (nwieters) [inline src/esm_runscripts/tidy.py]:
```suggestion
                f"    clean_this_rundir: {config["general"]["clean_this_rundir"]}"
```

File: src/esm_runscripts/tidy.py [hunk 38]
```diff
@@ -147,46 +147,48 @@ def _clean_run_determine_user_choice(config):
...
+            logger.error("general:")
+            logger.error(
                 "    clean_old_rundirs_except: ",
                 config["general"]["clean_old_rundirs_except"],
             )
-            print("    clean_runs: ", user_clean)
-            print()
-            print("Please only use one of these!")
-            print("------------------------------------------")
+            logger.error("    clean_runs: ", user_clean)
```
Comment (nwieters) [inline src/esm_runscripts/tidy.py]:
```suggestion
            logger.error(f"    clean_runs: {user_clean}")
```

File: src/esm_runscripts/tidy.py [hunk 39]
```diff
@@ -147,46 +147,48 @@ def _clean_run_determine_user_choice(config):
...
-        print("However, only the following types are valid:")
-        print("   * boolean")
-        print("   * integer (greater or equal to 0!)")
-        print("Please correct that")
-        print("------------------------------------------")
+        logger.error("------------------------------------------")
+        logger.error("Type Error!")
+        logger.error("You have set this in your config:")
+        logger.error("general:")
+        logger.error("    clean_runs: ", user_clean)
```
Comment (nwieters) [inline src/esm_runscripts/tidy.py]:
```suggestion
        logger.error(f"    clean_runs: {user_clean}")
```

File: src/esm_runscripts/tidy.py [hunk 40]
```diff
@@ -147,46 +147,48 @@ def _clean_run_determine_user_choice(config):
...
+        logger.error()
+        logger.error("This is of type: ", type(user_clean))
```
Comment (nwieters) [inline src/esm_runscripts/tidy.py]:
```suggestion
        logger.error(f"This is of type: {type(user_clean)}")
```

**Review by nwieters — CHANGES_REQUESTED** [review]
Hi @pgierz ,

thank you for starting this. Very nice work.

I found some calls of logger.info/debug/error that didn't really work, because there were commas in the calls separating the text. I guess this comes from the replaced print statements. But everything behind the first comma is not text but another argument to logger and not appearing in the log output.

I also found some flush=True arguments which are probably not logger arguments right? In some cases you also removed them.

I hope I found everything.

File: src/esm_runscripts/compute.py (hunk 2 above)
Comment (pgierz) [reply src/esm_runscripts/compute.py]: Yes, `logger.info` is what used to be be `print`, and `logger.debug` is what used to be print in verbose mode

File: src/esm_runscripts/filelists.py (hunk 17 above)
Comment (mandresm) [reply src/esm_runscripts/filelists.py:60]: Nadine is right, `print_datatime` has only a `logger.info` command, that means is going to always print independently of whether it's verbose or not.

File: src/esm_runscripts/filelists.py (hunk 19 above)
Comment (mandresm) [reply src/esm_runscripts/filelists.py]: Indeed

File: src/esm_runscripts/filelists.py [hunk 41]
```diff
@@ -911,10 +905,8 @@ def resolve_symlinks(config, file_source):
 
 
 def copy_files(config, filetypes, source, target):
-    if config["general"]["verbose"]:
-        print("\n::: Copying files", flush=True)
-        helpers.print_datetime(config)
-
+    logger.debug("\n::: Copying files")
+    helpers.print_datetime(config)
```
Comment (mandresm) [inline src/esm_runscripts/filelists.py:905]: outside verbose if statement?

File: src/esm_runscripts/filelists.py [hunk 42]
```diff
@@ -941,19 +933,15 @@ def copy_files(config, filetypes, source, target):
...
-                    if config["general"]["verbose"]:
-                        print(flush=True)
-                        print(f"::: copying file category: {category}")
-                        print(f"- source: {file_source}", flush=True)
-                        print(f"- target: {file_target}", flush=True)
-                        helpers.print_datetime(config)
+                    logger.debug(f"::: copying file category: {category}")
+                    logger.debug(f"- source: {file_source}")
+                    logger.debug(f"- target: {file_target}")
+                    helpers.print_datetime(config)
```
Comment (mandresm) [inline src/esm_runscripts/filelists.py:935]: outside verbose if statement?

File: src/esm_runscripts/filelists.py [hunk 43]
```diff
@@ -941,19 +933,15 @@ def copy_files(config, filetypes, source, target):
...
-                        if config["general"]["verbose"]:
-                            print(
-                                f"Source and target paths are identical, skipping {file_source}",
-                                flush=True,
-                            )
-                            helpers.print_datetime(config)
+                        logger.debug(
+                            f"Source and target paths are identical, skipping {file_source}",
+                        )
+                        helpers.print_datetime(config)
```
Comment (mandresm) [inline src/esm_runscripts/filelists.py:940]: outside verbose if statement?

File: src/esm_runscripts/filelists.py [hunk 44]
```diff
@@ -965,30 +953,26 @@ def copy_files(config, filetypes, source, target):
...
-                                if config["general"]["verbose"]:
-                                    print(
-                                        f"Source and target file are identical, skipping {file_source}",
-                                        flush=True,
-                                    )
-                                    helpers.print_datetime(config)
+                                logger.debug(
+                                    f"Source and target file are identical, skipping {file_source}",
+                                )
+                                helpers.print_datetime(config)
```
Comment (mandresm) [inline src/esm_runscripts/filelists.py:964]: outside verbose if statement?

File: src/esm_runscripts/logfiles.py [hunk 45]
```diff
@@ -117,7 +121,7 @@ class RuntimeLogger(_io.TextIOWrapper):
     """
     Logger object that takes care of both, writing the stdout and stderr to the
     'mini-log' file corresponding to the different elements of the workflow (observe,
-    tidy, ...), and writing to the system stdout/stderr so that it is catched by SLURM
+    tidy, ...), and writing to the system stdout/stderr so that it is caught by SLURM
```
Comment (mandresm) [inline src/esm_runscripts/logfiles.py:124]: Nice catch. You caught my bad English ;)

File: src/esm_runscripts/prepcompute.py [hunk 46]
```diff
@@ -40,9 +41,9 @@ def compile_model(config):
     if not version:
         return config
     if config.get("general", {}).get("run_number") == 1:
-        print("First year, checking if we need to compile...")
+        logger.info("First year, checking if we need to compile...")
         if not config.get("general", {}).get("use_compiled_model", True):
-            print(f"Huhu --> compiling {model}-{version}")
+            logger.debug(f"Huhu --> compiling {model}-{version}")
```
Comment (mandresm) [inline src/esm_runscripts/prepcompute.py]:
```suggestion
            logger.info(f"Huhu --> compiling {model}-{version}")
```
Shouldn't this be `info`?

File: src/esm_runscripts/prepcompute.py [hunk 47]
```diff
@@ -171,10 +167,9 @@ def modify_namelists(config):
     # Load and modify namelists:
 
     if config["general"]["verbose"]:
-        print("\n" "- Setting up namelists for this run...")
+        logger.debug("\n" "- Setting up namelists for this run...")
         for index, model in enumerate(config["general"]["valid_model_names"]):
-            print(f'{index+1}) {config[model]["model"]}')
-        print()
+            logger.debug(f'{index+1}) {config[model]["model"]}')
```
Comment (mandresm) [inline src/esm_runscripts/prepcompute.py]: Don't you need to get rid of the `config["general"]["verbose"]`.

File: src/esm_runscripts/prepcompute.py [hunk 48]
```diff
@@ -188,7 +183,7 @@ def modify_namelists(config):
         )
 
     if config["general"]["verbose"]:
-        print("::: end of namelist section\n")
+        logger.debug("::: end of namelist section\n")
```
Comment (mandresm) [inline src/esm_runscripts/prepcompute.py]: Don't you need to get rid of the `config["general"]["verbose"]`.

File: src/esm_runscripts/tidy.py [hunk 49]
```diff
@@ -147,46 +147,48 @@ def _clean_run_determine_user_choice(config):
...
-            print("Please only use one of these!")
-            print("------------------------------------------")
+            logger.error("------------------------------------------")
+            logger.error("You have set both in your config:")
+            logger.error()
+            logger.error("general:")
+            logger.error(
+                "    clean_this_rundir: ", config["general"]["clean_this_rundir"]
+            )
+            logger.error("    clean_runs: ", user_clean)
```
Comment (mandresm) [inline src/esm_runscripts/tidy.py]:
```suggestion
            logger.error(f"    clean_runs: {user_clean}")
```

**Review by mandresm — CHANGES_REQUESTED** [review]
Thanks @pgierz! Great work! Also @nwieters for a great review!

I added some other suggestions :)

File: src/esm_runscripts/prepcompute.py (hunk 46 above)
Comment (nwieters) [reply src/esm_runscripts/prepcompute.py]: Replaced debug by info

File: src/esm_runscripts/prepcompute.py (hunk 47 above)
Comment (nwieters) [reply src/esm_runscripts/prepcompute.py]: Removed `config["general"]["verbose"]`

File: src/esm_runscripts/prepcompute.py (hunk 48 above)
Comment (nwieters) [reply src/esm_runscripts/prepcompute.py]: Removed `config["general"]["verbose"]`

File: src/esm_runscripts/filelists.py (hunk 41 above)
Comment (nwieters) [reply src/esm_runscripts/filelists.py:905]: The `logger.info` call in `print_datetime` function was behind an if-statement referring to the variable `verbose_datetime_info` which does not exists anymore. So I deleted the if-statement and changed `logger.info` to `logger.debug` in the function `print_datetime`.

File: src/esm_runscripts/oasis.py (hunk 30 above)
Comment (nwieters) [reply src/esm_runscripts/oasis.py:170]: Does `logger.error` raises such an `sys.exit(2)` or is it obsolete here? 
I am asking because there are other `logger.error` calls still followed by such a `sys.exit(2)` call.

**Comment by nwieters** [comment]
Hi, I think I fixed all logger calls. From my side it is ready to be merged. There is just one change by @pgierz I am not sure about. The deletion of a `sys.exit(2)` call in oasis.py. I made a remark in the review. 
@mandresm and @pgierz maybe you can have a look at it again.

**Comment by nwieters** [comment]
Just found one small bug that broke the tests.

File: src/esm_runscripts/oasis.py (hunk 30 above)
Comment (mandresm) [reply src/esm_runscripts/oasis.py:170]: Fixed!

**Review by mandresm — APPROVED** [review]

**Comment by mandresm** [comment]
#bump
